packages/iTwinUI-react/src/core/Table/hooks/useResizeColumns.tsx

```diff
@@ -348,6 +352,19 @@ const isNewColumnWidthsValid = <T extends Record<string, unknown>>(
     }
   }
 
+  if (instance && instance.resizeMode === 'current-column') {
+    let newTableWidth = 0;
+    for (const header of instance.flatHeaders) {
+      newTableWidth += columnWidths[header.id]
+        ? columnWidths[header.id]
+        : getHeaderWidth(header);
+    }
+    // `tableWidth` is whole number therefore we need to round the `newTableWidth`
+    if (Math.round(newTableWidth) < instance.tableWidth) {
```

Comment:
Why we are not allowing to decrese table width?

Reply:
I don't like the idea that table will be smaller because that can cause some layout shifts or they can even break without devs even thinking about that. Also I checked other libs and although they allow to reduce column size but table size remains the same but their table columns are a bit different, they don't have strict borders.
![image](https://user-images.githubusercontent.com/36186912/191900608-a5a4b42f-68dc-4d4d-adcf-fd0fa182e227.png)

Reply:
I agree that it could break some layouts.
But users can reduce column width to get table to original size if other column is expanded. So not being able to decrease size of column in expand mode might confuse users and they could think it's a bug. 
Maybe we could consult UX team about expected behaviour or just talk within a team.